table/server/common/src/main/java/alluxio/table/common/udb/UdbFilterSpec.java

```diff
@@ -0,0 +1,505 @@
+            partitionNameTemplates);
+        // forbid empty spec
+        Preconditions.checkArgument(!partitionNameTemplates.isEmpty(),
+            "Empty partition specification set for table %s",
+            tableName);
+        // forbid NONE mode
+        Preconditions.checkArgument(partitionMode != Mode.NONE,
+            "Invalid mode %s for partitions of table %s, must be either include or exclude",
+            partitionMode, tableName);
+        // forbid both exact name and partition spec for a table at the same time
+        Preconditions.checkArgument(mBypassedTables.stream()
+                .noneMatch(template -> template instanceof ExactNameTemplate
+                    && template.matches(tableName)),
+            "Table %s is already specified with exact name, "
+                + "cannot have partition specification at the same time",
+            tableName);
+      }
+    }
+  }
+
+  public static UdbFilterSpec empty() {
+    return EMPTY_INSTANCE;
+  }
+
+  /**
+   * Checks if a table should be bypassed.
+   *
+   * @param tableName the table name
+   * @return true if the table is configured to be bypassed, false otherwise
+   * @see UdbFilterSpec#isFullyBypassedTable(String)
+   */
+  public boolean isBypassedTable(String tableName) {
+    boolean isSpecifiedByName = matchedByTemplates(mBypassedTables, tableName);
+    if (isSpecifiedByName) {
+      if (mTableBypassMode == Mode.EXCLUDE) {
+        return false;
+      }
+      return shadowedByIgnore(tableName, true);
+    }
+    boolean hasBypassedPartition = mBypassedPartitions.containsKey(tableName);
+    if (hasBypassedPartition) {
+      // no need to check mTableBypassMode here, bc if a table has partition specifications,
+      // it must be in inclusion mode (or none)
+      return shadowedByIgnore(tableName, true);
+    }
+    // the table is not listed in any way
+    if (mTableBypassMode == Mode.EXCLUDE) {
+      return shadowedByIgnore(tableName, true);
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Checks if all partitions of a table should be bypassed.
+   *
+   * @param tableName the table name
+   * @return true if the table is configured to be fully bypassed, false otherwise
+   * @see UdbFilterSpec#isBypassedTable(String)
+   */
+  public boolean isFullyBypassedTable(String tableName) {
+    boolean isFullyBypassed = matchedByTemplates(mBypassedTables, tableName);
+    return shadowedByIgnore(tableName, handleInclusionExclusion(mTableBypassMode, isFullyBypassed));
+  }
+
+  /**
+   * Checks if a table is ignored.
+   *
+   * @param tableName the table name
+   * @return true if the table is ignored, false otherwise
+   */
+  public boolean isIgnoredTable(String tableName) {
+    boolean isIgnored = matchedByTemplates(mIgnoredTables, tableName);
+    return handleInclusionExclusion(mIgnoreMode, isIgnored);
+  }
+
+  /**
+   * Checks by a partition's name if it should be bypassed.
```

Comment:
```suggestion
   * Check whether it is a bypassed partition by name.
```

Reply:
Fixed in cef77153696b4028a525829d6d17d898fa972e6c. Removed unnecessary mention of "by name" in description as it is unambiguous given the parameters' description.